patroni/config.py

```diff
@@ -120,12 +120,32 @@ def dynamic_configuration(self):
     def check_mode(self, mode):
         return bool(parse_bool(self._dynamic_configuration.get(mode)))
 
+    def _load_config_path(self, path):
+        """
+        If path is a file, loads the yml file pointed to by path.
+        If path is a directory, loads all yml files in that directory in alphabetical order
+        """
+        if os.path.isfile(path):
+            files = [path]
+        elif os.path.isdir(path):
+            files = [f for f in sorted(os.listdir(path))
+                     if (f.endswith('.yml') or f.endswith('.yaml')) and os.path.isfile(f)]
+        else:
+            logger.error('config path %s is neither directory nor file', path)
+            raise ConfigParseError('invalid config path')
```

Comment:
Could you please also add the test-case for these lines?

Reply:
Technically this is unreachable because of the `os.path.exists(configfile)` check in `__init__`. I don't think there are any other possibilities for an existing path. I kept it there for safety in case something changes to the code in the future that makes it suddenly reachable. Would you rather that I remove the whole else part?

Reply:
It is perfectly reachable:
```python
In [1]: import os

In [2]: os.path.isfile('data/postgresql0/.s.PGSQL.5432')
Out[2]: False

In [3]: os.path.isdir('data/postgresql0/.s.PGSQL.5432')
Out[3]: False
```

`data/postgresql0/.s.PGSQL.5432` is neither file nor directory, but unix-socket.

Reply:
Ah, I didn't know that. Ok, I've added a test case that asserts it throws when an existing path is neither a file nor a directory